SoftLayer/managers/ordering.py

```diff
@@ -356,8 +357,10 @@ def get_price_id_list(self, package_keyname, item_keynames):
             # can take that ID and create the proper price for us in the location
             # in which the order is made
             if matching_item['itemCategory']['categoryCode'] != "gpu0":
-                price_id = [p['id'] for p in matching_item['prices']
-                            if not p['locationGroupId']][0]
+                price_id = None
```

Comment:
Lets move all the logic for finding a price_id into the `get_item_price_id` method. 

```python
price_id = self.get_item_price_id(core, matching_item['prices'])
```


Then the `get_item_price_id` would be like this

```python
    @staticmethod
    def get_item_price_id(core, prices):
        """get item price id"""
        price_id = None
        for price in prices:
            if not price['locationGroupId']:
                capacity_min = int(price.get('capacityRestrictionMinimum', -1))
                capacity_max = int(price.get('capacityRestrictionMaximum', -1))
                if capacity_min == -1:
                    price_id = price['id']
                elif capacity_min <= int(core) <= capacity_max:
                    price_id = price['id']
        return price_id
```

```shell
./slcli order place --verify --billing hourly SUSPEND_CLOUD_SERVER DALLAS12 --preset B1_8X16X100 BANDWIDTH_0_GB_2 1_GBPS_PRIVATE_NETWORK_UPLINK REBOOT_REMOTE_CONSOLE 1_IP_ADDRESS UNLIMITED_SSL_VPN_USERS_1_PPTP_VPN_USER_PER_ACCOUNT OS_WINDOWS_2012_R2_FULL_STD_64_BIT MONITORING_HOST_PING NOTIFICATION_EMAIL_AND_TICKET AUTOMATED_NOTIFICATION NESSUS_VULNERABILITY_ASSESSMENT_REPORTING --complex-type SoftLayer_Container_Product_Order_Virtual_Guest --extras '{"hardware": [{"hostname": "test", "domain": "us.test.net"}]}'
BANDWIDTH_0_GB_2                                     0 GB Bandwidth Allotment                               0
1_GBPS_PRIVATE_NETWORK_UPLINK                        1 Gbps Private Network Uplink                          0
REBOOT_REMOTE_CONSOLE                                Reboot / Remote Console                                0
1_IP_ADDRESS                                         1 IP Address                                           0
UNLIMITED_SSL_VPN_USERS_1_PPTP_VPN_USER_PER_ACCOUNT  Unlimited SSL VPN Users & 1 PPTP VPN User per account  0
OS_WINDOWS_2012_R2_FULL_STD_64_BIT                   Windows Server 2012 R2 Standard Edition (64 bit)       0
MONITORING_HOST_PING                                 Host Ping                                              0
NOTIFICATION_EMAIL_AND_TICKET                        Email and Ticket                                       0
AUTOMATED_NOTIFICATION                               Automated Notification                                 0
NESSUS_VULNERABILITY_ASSESSMENT_REPORTING            Nessus Vulnerability Assessment & Reporting            0
GUEST_DISK_100_GB_SAN                                100 GB (SAN)                                           0
GUEST_CORES_8                                        8 x 2.0 GHz or higher Cores                            0
RAM_16_GB_HDD                                        16 GB                                                  0

```

Reply:
Ok, thank you for the suggestion, I will refactor it.

Reply:
Done